packages/material-ui/src/TextField/TextField.d.ts

```diff
@@ -66,6 +67,10 @@ export interface BaseTextFieldProps
    * Props applied to the [`InputLabel`](/api/input-label/) element.
    */
   InputLabelProps?: Partial<InputLabelProps>;
+  /**
+   * [Attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Attributes) applied to the `input` element.
+   */
+  inputProps?: InputBaseProps['inputProps'];
```

Comment:
All variations of TextField pointed to `InputBaseProps`. Makes the interface of each variation smaller and helps identify what we actually gain from discriminant unions (hint: not much but will discuss in a later PR).